docs/api.md

```diff
@@ -7,102 +7,167 @@ method arguments and return values.
-***This method requires being logged in as an administrator***
+Problem management allows access to the various problems that exist on the website.
+Problems will always be described using the problem JSON object:
 
-__URL:__ `/api/problems/`
+__[Problem Object](#the-problem-object)__
 
-__Method:__ `POST`
+Problem objects contain an array of sample cases objects, which are used to describe
+an example mapping of input to output for a problem:
 
-__Form Data:__
+__[Sample Case Object](#the-sample-case-object)__
 
-| Title | Required | Form Name | Description|
-| --- | --- | --- | --- |
-| Problem Name | __Required__ | `name` | Full problem name (<= 32 characters) |
-| Description | __Required__ | `description` | Problem background story |
-| Input Description | __Required__ | `input_desc` | Description of input values |
-| Output Description | __Required__ | `output_desc` | Description of output values |
-| Sample Case(s) | __Required__ | `cases` | JSON array of sample input/output (see below) |
-| Time Limit | Optional | `time_limit` | Max execution time (in seconds) |
-| Difficulty | Optional | `difficulty` | 0-100 difficulty rating |
-| Appeared In | Optional | `appeared_in` | String name of original competition |
-| Input Files | __Required__ | `in_file` | Zipped (.zip) directory of all input files |
-| Output Files | __Required__ | `out_file` | Zipped (.zip) directory of all output files |
-| Solution File | __Required__ | `sol_file` | Solution program (not zipped) |
 
-For __Sample Cases__, must be a string JSON array of mappings `input` and `output`.
-For example,
+From here, any user can perform the following actions:
+
+1. [Get Data on all Problems](#get-data-on-all-problems)
+2. [Get Data on a Specific Problem](#get-data-on-a-problem)
+
+Those logged in as an administrator can perform additional actions:
+
+3. [Create a New Problem](#create-a-new-problem)
+4. [Edit an Existing Problem](#edit-an-existing-problem)
+5. [Delete a Problem](#delete-a-problem)
+
+
+## Problem Management Objects
+
+### The Problem Object
+
+| Name | Type | Description |
+| --- | --- | --- |
+|`added`|`int`|The UTC timestamp indicating when this problem was uploaded|
+|`appeared`|`String`|The contest this problem originally appeared in|
+|`comp_release`|`int`|The competition `cid` for the competition this was used in|
+|`description`|`String`|The problem's description|
+|`difficulty`|`int`|The difficulty of the problem from 0 (easiest) to 100 (hardest)|
+|`input_desc`|`String`|A description of the problem's input|
+|`name`|`String`|The name of the problem|
+|`output_desc`|`String`|A description of the problem's output|
+|`pid`|`int`|The problem's unique id.|
+|`sample_cases`|`Sample Case[]`|A list of samples cases, as described below|
+|`shortname`|`String`|A unique string with no spaces used to identify the problem|
+
+The `sample_cases` field consists of an array of [Sample Case](#the-sample-case-object) objects.
+
+
+__Example:__
 
 ```json
-[
-  {
+{
```

Comment:
Several of the fields here are either null or empty. For sample data, it makes sense to fill it all out so that the reader of the documentation can actually understand what it will look like.